test/CMakeLists.txt

```diff
@@ -39,4 +39,15 @@ if (WITH_TEST_GENERATOR)
     add_subdirectory(generator)
 endif ()
 
+if(MSVC)
+    set(ENABLE_RUNTIME_TESTS OFF)
```

Comment:
Why can't we do the runtime tests under MSVC?

Reply:
Because the runtime tests include src/runtime/runtime_internal.h which was written to only support clang (GCC's front end is close enough it works fine as well).  I originally setup the tests to compile with clang (in the same way as the actual runtime bitcode files), but that wasn't very clean and didn't integrate well with the other tests, so we switched to just using the native system compiler.  We could certainly add MSVC compatibility to runtime_internal.h and ifdef the attributes and types that are causing compile errors, but it seemed outside the scope of this PR.

Reply:
Fair, maybe worth capturing this in a comment there for posterity

Reply:
I would actually print something to the configure log via `message(STATUS)`

Also, a user can still write `-DWITH_TEST_RUNTIME=YES` at the command line with MSVC and enable these tests. Is that intentional? If not, you might want to write something like:

```
include(CMakeDependentOption)  # if necessary, might be included in a file above this and so would be redundant
cmake_dependent_option(WITH_TEST_RUNTIME "Build runtime tests" ON
                       "NOT MSVC" OFF)
if (WITH_TEST_RUNTIME)
  message(STATUS "Halide internal runtime tests enabled")
  add_subdirectory(runtime)
else ()
  message(STATUS "Halide internal runtime tests disabled")
endif ()
```

Reply:
@derek-gerstmann - That's what I'm here for :slightly_smiling_face: